### PR TITLE
[#232]refactor config module to support diffirent configuration implementations

### PR DESCRIPTION
**Pull request by GuoYL123**

### CODE REVIEW

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  public ConfigCenterClient(String urls,
+      HttpTransport httpTransport) {
+    super(urls, httpTransport);
+  }
+
+  public Map<String, String> loadAll(ServiceCombConfigProperties serviceCombConfigProperties,
+      String project) throws RemoteOperationException {
+    project = project != null && !project.isEmpty() ? project : ConfigConstants.DEFAULT_PROJECT;
+    String dimensionsInfo = spliceDimensionsInfo(serviceCombConfigProperties);
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java:34]: Can be replace with StringUtils.isEmpty

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      throw new RemoteOperationException("read response failed. " + response, e);
+    }
+  }
+
+
+  private String spliceDimensionsInfo(ServiceCombConfigProperties serviceCombConfigProperties) {
+    String result =
+        serviceCombConfigProperties.getServiceName() + ConfigConstants.DEFAULT_APP_SEPARATOR
+            + serviceCombConfigProperties.getAppName();
+    if (serviceCombConfigProperties.getVersion() != null && !serviceCombConfigProperties
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java:99]: replace with StringUtils.isEmpty

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClient.java [hunk 3]
```diff
@@ -77,171 +52,6 @@ public ServiceCombConfigClient(String urls, HttpTransport httpTransport) {
    */
   public Map<String, String> loadAll(ServiceCombConfigProperties serviceCombConfigProperties,
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClient.java:53]: loadAll can be declare abstract

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/model/KVBody.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package com.huaweicloud.config.kie;
+package com.huaweicloud.config.model;
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/model/KVBody.java:18]: This model is only used for kie. So I think you do not give a properpy direrctory structure to hold different implementations.

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClientBuilder.java [hunk 5]
```diff
@@ -45,14 +47,19 @@ public ServiceCombConfigClientBuilder setServiceCombSSLProperties(
...
+    this.serviceCombConfigProperties = serviceCombConfigProperties;
     return this;
   }
 
   public ServiceCombConfigClient createServiceCombConfigClient() {
     DefaultHttpTransport httpTransport = DefaultHttpTransport.getInstance(serviceCombSSLProperties);
     httpTransport.setServiceCombAkSkProperties(serviceCombAkSkProperties);
-    return new ServiceCombConfigClient(url, httpTransport);
+    if (!StringUtils.isEmpty(serviceCombConfigProperties.getServerType())
+        && serviceCombConfigProperties.getServerType().equals("kie")) {
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClientBuilder.java:60]: You can use "kie".equalsIgnoreCase to make logic simpler.

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        properties.load(new StringReader(kvDoc.getValue()));
+      } else if (vtype == (ValueType.text) || vtype == (ValueType.string)) {
+        kvMap.put(kvDoc.getKey(), kvDoc.getValue());
+        return kvMap;
+      } else {
+        kvMap.put(kvDoc.getKey(), kvDoc.getValue());
+        return kvMap;
+      }
+      kvMap = toMap(kvDoc.getKey(), properties);
+      return kvMap;
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java:167]: This code is very complicated, you can use some desgin patterns to make it simpler, such Factory based on valueType.

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+          }
+        }
+      }
+    }
+    //kv is priority
+    for (KVDoc kvDoc : appList) {
+      resultMap.putAll(processValueType(kvDoc));
+    }
+    for (KVDoc kvDoc : serviceList) {
+      resultMap.putAll(processValueType(kvDoc));
```
Comment (liubao68) [inline spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java:134]: This code is very long, and seams you process serviceList, versionList, etc with the same code `resultMap.putAll(processValueType(kvDoc));` , and the logic maybe better refactored.

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java (hunk 1 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java:34]: done

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java (hunk 2 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ConfigCenterClient.java:99]: done

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClient.java (hunk 3 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClient.java:53]: done

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClientBuilder.java (hunk 5 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/ServiceCombConfigClientBuilder.java:60]: very useful

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java (hunk 6 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java:167]: using switch to simplify

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java (hunk 7 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java:134]: seems not easy to refactory

File: spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/model/KVBody.java (hunk 4 above)
Comment (GuoYL123) [reply spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/model/KVBody.java:18]: I think model is better than kie , it's hard to call a higher better name

**Pull request merged**